### ISSUE TITLE
fulltext-search - circular reference

### ISSUE THREAD

**Issue by nelsongonzalez** [firebase/functions-samples#104]
Hi,

I run the fulltext-search sample an there is a potential circular reference.

Te sample code looks like:
```javascript
// Updates the search index when new blog entries are created or updated.
exports.indexentry = functions.database.ref('/blog-posts/{blogid}').onWrite(event => {
  const index = client.initIndex(ALGOLIA_POSTS_INDEX_NAME);
  const firebaseObject = event.data.val();
  firebaseObject.objectID = event.data.key;

  return index.saveObject(firebaseObject).then(
      () => event.data.adminRef.child('last_index_timestamp').set(
          Date.parse(event.timestamp)));
});
```
You could experiment a circular reference:

1. You write a path `/blog-posts/{blogid}`
1. The function `exports.indexentry` will finish writing a child called `last_index_timestamp`
1. Then, because the child `/blog-posts/{blogid}/last_index_timestamp` was added, the function `exports.indexentry` will be called another time, and so on

I expect it helps to fix the sample.

Thanks.

**Comment by nicolasgarnier** [comment]
Fixed! Thanks for the report! We now have a trigger on `/blog-posts/{blogid}/text` instead.

**Comment by nelsongonzalez** [comment]
Hi,

The circular reference persist, the only diference is that we have changed the path:
1. You write a path `/blog-posts/{blogid}/text`
1. The function `exports.indexentry` will finish writing a child called `last_index_timestamp`
1. Then, because the child `/blog-posts/{blogid}/text/last_index_timestamp` was added, the function `exports.indexentry` will be called another time, and so on

I think a possible solution could be using the attribute `event.data.adminRef.parent`, for example:
```javascript
return index.saveObject(firebaseObject).then(
      () => event.data.adminRef.parent.child('last_index_timestamp').set(
          Date.parse(event.timestamp)));
});
```

Using the attribute `parent`:
1. You write a path `/blog-posts/{blogid}/text`
1. The function `exports.indexentry` will finish writing a path called `/blog-posts/{blogid}/last_index_timestamp`
1. Then, because the path `/blog-posts/{blogid}/last_index_timestamp` is not a child of `/blog-posts/{blogid}/text` the function `exports.indexentry` will not be called another time

A similar firebase sample [here](https://github.com/firebase/functions-samples/blob/master/quickstarts/uppercase/functions/index.js#L43-L59)

I hope it helps.

**Comment by nicolasgarnier** [comment]
Yes I forgot to use the parent here. Fixed in acaec08

btw it might be simpler sometimes for you to fix the file yourself using a Pull Request ;) Just don't hesitate to send PRs if you see an easy fix :)

**Comment by nelsongonzalez** [comment]
Thanks Nicolas, I Will do that the next time. I am new in the world of
collaborating with code but I like the idea.
